packages/types/src/ComponentThemeMap.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 - present Instructure, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import { ThemeVariables } from './ComponentThemeVariables'
+
+type ComponentThemeVariables = {
+  Avatar: ThemeVariables['Avatar']
+  Alert: ThemeVariables['Alert']
+  Badge: ThemeVariables['Badge']
+  Billboard: ThemeVariables['Billboard']
+  Breadcrumb: ThemeVariables['Breadcrumb']
+  BaseButton: ThemeVariables['BaseButton']
+  Button: ThemeVariables['BaseButton']
+  CloseButton: ThemeVariables['CloseButton']
+  CondensedButton: ThemeVariables['BaseButton']
+  IconButton: ThemeVariables['BaseButton']
+  Byline: ThemeVariables['Byline']
+  'Calendar.Day': ThemeVariables['CalendarDay']
+  Calendar: ThemeVariables['Calendar']
+  CheckboxFacade: ThemeVariables['CheckboxFacade']
+  ToggleFacade: ThemeVariables['ToggleFacade']
+  CodeEditor: ThemeVariables['CodeEditor']
+  'DrawerLayout.Content': ThemeVariables['DrawerLayoutContent']
+  'DrawerLayout.Tray': ThemeVariables['DrawerLayoutTray']
```

Comment:
You are right, they could be using these components directly, I only used the `componentId` property for these kind of components, but it would be better to list all of the possible variants.